app/controllers/ManageTeamMemberController.scala

```diff
@@ -55,7 +57,28 @@ class ManageTeamMemberController @Inject()(
   def showAllTeamMembers: Action[AnyContent] = Action.async { implicit request =>
     isAuthorisedAgent { arn =>
       isOptedIn(arn) { _ =>
-        Ok(s"showAllTeamMembers not yet implemented $arn").toFuture
+        groupService.getTeamMembers(arn)().flatMap { maybeTeamMembers =>
+          val searchFilter: SearchFilter = SearchAndFilterForm.form().bindFromRequest().get
+          searchFilter.submit.fold(
+            //no filter/clear was applied
+            Ok(manage_team_members_list(teamMembers = maybeTeamMembers, form = SearchAndFilterForm.form()))
+          )({
+            //clear/filter buttons pressed
+            case "clear" =>
+              Redirect(routes.ManageTeamMemberController.showAllTeamMembers)
+            case "filter" =>
+              val filterByName = maybeTeamMembers.getOrElse(Seq.empty)
```

Comment:
ideally this logic would be in a service but we could get to that later

Reply:
Yes ideally I'd move it into a service, want to refactor later to make better use of our existing logic in group service

To do that first we should separate the filtering out from the added logic needed for handling checkboxes

Reply:
okay